tests/st/cycle_gan.py

```diff
@@ -0,0 +1,252 @@
+                                                                                    mean_class_iou))
+    reporter.info("************ Per class numbers below ************")
+    for i, cl in enumerate(CS.classes):
+        while len(cl) < 15:
+            cl = cl + ' '
+        reporter.info("{}: acc = {}, iou = {}".format(cl, per_class_acc[i], per_class_iou[i]))
+    reporter.end_eval()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='MindSpore Cycle GAN Example')
+    parser.add_argument('--device_target', type=str, default="CPU", choices=['Ascend', 'GPU', 'CPU'],
+                        help='device where the code will be implemented (default: CPU)')
+    parser.add_argument('--dataset_path', type=str, default=None, help='cityscape dataset path.')
+    parser.add_argument('--phase', type=str, default="train", help='train, eval or predict.')
+    parser.add_argument('--model', type=str, default="resnet", choices=("resnet", "unet"),
+                        help='generator model, should be in [resnet, unet].')
+    parser.add_argument('--max_epoch', type=int, default=200, help='epoch size for training, default is 200.')
+    parser.add_argument('--n_epoch', type=int, default=100,
+                        help='number of epochs with the initial learning rate, default is 100')
+    parser.add_argument('--batch_size', type=int, default=1, help='Batch size.')
+    parser.add_argument("--save_checkpoint_epochs", type=int, default=10,
+                        help="Save checkpoint epochs, default is 10.")
+    parser.add_argument("--G_A_ckpt", type=str, default=None, help="pretrained checkpoint file path of G_A.")
+    parser.add_argument("--G_B_ckpt", type=str, default=None, help="pretrained checkpoint file path of G_B.")
+    parser.add_argument("--D_A_ckpt", type=str, default=None, help="pretrained checkpoint file path of D_A.")
+    parser.add_argument("--D_B_ckpt", type=str, default=None, help="pretrained checkpoint file path of D_B.")
+    parser.add_argument('--outputs_dir', type=str, default='./outputs',
+                        help='models are saved here, default is ./outputs.')
+    parser.add_argument('--save_imgs', type=bool, default=True,
+                        help='whether save imgs when epoch end, default is True.')
+    parser.add_argument("--cityscapes_dir", type=str, help="Path to the original cityscapes dataset")
+    parser.add_argument("--result_dir", type=str, help="Path to the generated images to be evaluated")
+    args_opt = parser.parse_args()
+
+    context.set_context(mode=context.GRAPH_MODE, device_target=args_opt.device_target)
+
+    dataset_path = args_opt.dataset_path
+    phase = args_opt.phase
+    G_A_ckpt = args_opt.G_A_ckpt
+    G_B_ckpt = args_opt.G_B_ckpt
+
+    if phase == "predict" and (G_A_ckpt is None or G_B_ckpt is None):
+        raise ValueError('Must set G_A_ckpt and G_B_ckpt in predict phase!')
+
+    if dataset_path is None and (phase in ["train", "predict"]):
```

Comment:
Please elaborate the difference between `dataset_path` and `cityscapes_dir` value.